### PR TITLE
ENH: support missing `properties` key in `from_features`

### LINKED ISSUE #2506: ENH: from_features - allow not passing in "properties"
https://github.com/geopandas/geopandas/blob/5c8ae832e030228b5deec02588c4e1b78dee0f18/geopandas/geodataframe.py#L651

It checks if `properties` is None. Any issues changing it to feature.get("properties") so this scenario could be supported:
```python
collection = {
    "type": "FeatureCollection",
    "features": [
        {
            "type": "Feature",
            "id": "095af6d6-bfcd-4cca-a72c-bf2efd1cf917",
            "geometry": {
                "type": "Point",
                "coordinates": [-10.0, -10.0]
            }
        },
        {
            "type": "Feature",
            "id": "a7ad51b6-ad2d-4f75-806a-d68ab9e07e5b",
            "geometry": {
                "type": "Point",
                "coordinates": [-10.1, -10.1]
            }
        }
    ]
}
gdf = geopandas.GeoDataFrame.from_features(collection["features"])
```

### PR DESCRIPTION
**Pull request by alinacherkas**
Hi! This fixes #2506, an old bug that resulted in a `KeyError` when running `from_features` on data without the `properties` field. I have also updated an old test to be clear that empty or `null` `properties` do not produce any column.

Let me know if anything needs to be adjusted.

### CODE REVIEW

File: geopandas/geodataframe.py [hunk 1]
```diff
@@ -816,9 +816,7 @@ def from_features(
                 "geometry": shape(feature["geometry"]) if feature["geometry"] else None
             }
             # load properties
-            properties = feature["properties"]
-            if properties is None:
-                properties = {}
+            properties = feature.get("properties") or {}
```
Comment (m-richards) [inline geopandas/geodataframe.py:818]:
```suggestion
            properties = feature.get("properties", {})
```

**Review by m-richards — APPROVED** [review]
Thanks @alinacherkas for this! Are you able to add a changelog entry for this, under a new Version 1.2 heading (currently the 1.1.1 heading is most recent, but that's unreleased and for regressions only, this would go into 1.2)

File: geopandas/geodataframe.py (hunk 1 above)
Comment (alinacherkas) [reply geopandas/geodataframe.py:818]: Thanks @m-richards , but this won't work if `properties` field is set to `None`. My one-liner handles both cases (the field is not there or it is there and it is set to `None`) without the need to check `if properties is None`.

**Comment by alinacherkas** [comment]
Hi @m-richards , just following up on this minor change to see if it can be merged now (not sure why one of the tests is failing, it seems to be an unrelated one).

**Comment by m-richards** [comment]
> Hi @m-richards , just following up on this minor change to see if it can be merged now (not sure why one of the tests is failing, it seems to be an unrelated one).

@alinacherkas yes the failure is unrelated and #3611 fixed that in main. We normally try to have two sets of eyes reviewing if we can, but this is pretty well defined and resolves a long standing feature request.
